runtime/runtime/src/actions.rs

```diff
@@ -971,24 +1004,34 @@ pub(crate) fn check_account_existence(
         }
         Action::Transfer(_) => {
             if account.is_none() {
-                return if config.wasm_config.implicit_account_creation
-                    && is_the_only_action
-                    && account_is_implicit(account_id, config.wasm_config.eth_implicit_accounts)
-                    && !is_refund
-                {
-                    // OK. It's implicit account creation.
-                    // Notes:
-                    // - The transfer action has to be the only action in the transaction to avoid
-                    // abuse by hijacking this account with other public keys or contracts.
-                    // - Refunds don't automatically create accounts, because refunds are free and
-                    // we don't want some type of abuse.
-                    // - Account deletion with beneficiary creates a refund, so it'll not create a
-                    // new account.
-                    Ok(())
-                } else {
-                    Err(ActionErrorKind::AccountDoesNotExist { account_id: account_id.clone() }
-                        .into())
-                };
+                return check_transfer_to_nonexisting_account(
+                    config,
+                    account_id,
+                    implicit_account_creation_eligible,
+                );
+            }
+        }
+        #[cfg(feature = "protocol_feature_nonrefundable_transfer_nep491")]
+        Action::NonrefundableStorageTransfer(_) => {
+            if account.is_none() {
+                return check_transfer_to_nonexisting_account(
+                    config,
+                    account_id,
+                    implicit_account_creation_eligible,
+                );
```

Comment:
Minor nit: If this is written as

```
Action::Transfer(_) if account.is_none() => { ... }
Action::NonrefundableStorageTransfer(_) if account.is_none() => { ... }
```

Then upon eventual stabilization these branches can be merged together like many others elsewhere in this PR.

Reply:
We have `else` branch for `Action::NonrefundableStorageTransfer(_)`. I think we cannot merge these match arms.

Reply:
Note the `if` guards in the example, so the overall post-merge state would look like this:

```rust
Action::Transfer(_) | Action::NonrefundableStorageTransfer(_) if account.is_none() => {
    /* the if branch */
}
Action::Transfer(_) => {}
Action::NonrefundableStorageTransfer(_) => { /* the else branch */ }
```

but either way a minor nit so fine to ignore it.

Reply:
I see, makes sense. Added TODO comment to merge it this way on stabilization.